src/util/string_utils.js

```diff
@@ -10,6 +10,11 @@ const email_regexp = /^((([a-z]|\d|[!#\$%&'\*\+\-\/=\?\^_`{\|}~]|[\u00A0-\uD7FF\
 const escape_regexp = /[\\^$.*+?()[\]{}|]/g;
 const access_key_regexp = /^[a-zA-Z0-9]{20}$/;
 const secret_key_regexp = /^[a-zA-Z0-9+/]{40}$/;
+// regex for IAM service
+const AWS_IAM_PATH_REGEXP = /(\u002F)|(\u002F[\u0021-\u007E]+\u002F)/;
```

Comment:
using unicode code numbers for ascii characters makes it harder to read. to use forward-slash in regex you can either use \/ or my preferred [/]

Reply:
and what is this range \u0021-\u007e? if this is needed then better document it and explain what this means.

Reply:
I took the pattern for the docs, for example ([CreateUser](https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateUser.html#API_CreateUser_RequestParameters)):

> Pattern: (\u002F)|(\u002F[\u0021-\u007E]+\u002F)

Explanation of the pattern:

> a string of characters consisting of either a forward slash (/) by itself or a string that must begin and end with forward slashes. In addition, it can contain any ASCII character from the ! (\u0021) through the DEL character (\u007F), including most punctuation characters, digits, and upper and lowercased letters.

I can add the link as an inline comment, will it be easier?

I didn't understand this comment:
> using unicode code numbers for ascii characters makes it harder to read. to use forward-slash in regex you can either use / or my preferred [/]

Would you please add a suggestion?
For the forward and backward slash I'm testing it in the code itself (the pattern doesn't include it):
`const valid_aws_path = input_path.startsWith('/') && input_path.endsWith('/') && AWS_IAM_PATH_REGEXP.test(input_path);
`

Reply:
ok nevermind, better leave it as as.